Documentation/proposal-runconfig_inside_osv.md

```diff
@@ -1,55 +1,286 @@
+ * what runtimes are available
+ * what runtime-specific attributes are supported
+
+Once she successfully composes the image, she takes the role of Application user.
+
+Example: NodeJS. User uses 'node' runtime (which downloads pre-prepared node package), but provides
+her own NodeJS JavaScript application code.
+
+##### Runtime developer
+Extends Capstan tool by implementing runtime.Runtime interface (Golang programmer). She must be
+provided with this information:
+ * what packages are available
+ * how to use package
+
+It must be as easy as possible to add new runtime since we may expect PRs for new runtimes!
+
+QUICK THOUGHT: Is it really neccessary that runtime is hard-coded in Capstan? Would it be better to
+support runtimes written in yaml and included in package itself? Not sure this is possible, though.
+
+##### Package developer
+Statically compiles new languages from source into shared objects and bundles them into Capstan
+package.
+
+NOTE: There is a mess with the hierarchy of packages. Would it make sense to introduce "package type"?
+We could have:
+ * *shared-object*: Packages that only provide shared objects
+ * *runtime*: Packages that provide runtime (may require multiple *shared-object*). These are
+ packages that doesn't work per-se; user needs to provide application code to compose with.
+ * *application*: Packages that user should just download, compose without modifications and run.
+    Only for this packages it's possible to provide the already-compiled image. These packages must
+    make use of Agent functionality.
+
+Role 'Package developer' is the one that prepares *shared-object* packages, role 'Runtime developer'
+prepares *runtime* packages and role 'Runtime user' prepares *application* packages. Role
+'Application user' depends on Runtime user to publish their application.
+
+
+
 ### Composing image
-File meta/run.yaml is present at package root directory (same as until now)
+When user runs `capstan package compose`, the content of meta/run.yaml file is processed and new
+files are generated and uploaded to the OSv image (assume we upload them to folder `/osv-runtime`):
 
-When user runs `capstan package compose` the content of meta/run.yaml file is processed and new files are generated inside the OSv image folder `/osv-runtime`:
+```
+user-friendly attributes of meta/run.yaml ---> complex run command for each named config
+```
 
-Capstan should create a separate file for each named configuration e.g.
+A separate file containing only run command string is created for each named configuration e.g.
 ```
 /osv-runtime
    |--burek
-   |--pizza   <--------\
-   |                    |
-   |--default-runtime  /
+   |--pizza
 ```
-Each runtime file should contain only one line representing the run command that is to be run inside OSv:
+Example:
 ```
 # content of file /osv-runtime/burek:
 node burek.js
 ```
+It is neccessary to use Capstan to generate run command since runtime user does not have enough
```

Comment:
I think that this is fine, but I wouldn't prevent users from adding a file directly into /osv-runtime instead of specifying the command by means of `meta/run.yaml`.